packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -330,6 +330,49 @@ const findNamedExportModuleNode = (
   };
 };
 
+/**
+ * Will recursively checks if identifier name is coming from destructuring. If yes,
+ * then will return the resolved identifer. We can look for identifier name
+ * either in destructuring key or its value.
+ *
+ * @param name Identifier name to resolve
+ * @param node Any Expression node
+ * @param resolveFor Either resolve destructuring key or its value
+ */
+export const resolveIdentifierComingFromDestructuring = ({
+  name,
+  node,
+  resolveFor = 'key',
+}: {
+  name: string;
+  node: t.Expression | undefined;
+  resolveFor?: 'key' | 'value';
```

Comment:
what if we just returned both?

Reply:
Yups we can 😃

Reply:
Another thought. Here we are traversing until we hit dead end. We either need to search for `key` or `value`. I mean where we will break the recursion. When either `key` is found or `value` is found?

Reply:
no worries then :)